app/full_text_search.py

```diff
@@ -171,10 +199,18 @@ def create_document(record_id, repo, **kwargs):
     fields.append(appengine_search.TextField(name='record_id', value=record_id))
 
     # Add name and location romanized by unidecode
+    name_fields = ['given_name', 'full_name', 'family_name', 'alternate_names']
     for field in kwargs:
         romanized_value = script_variant.romanize_word(kwargs[field])
         fields.append(
             appengine_search.TextField(name=field, value=romanized_value))
+        if romanized_value and field in name_fields:
+            connected_romanized_value = ''
+            for x in range(5):
```

Comment:
This code is very similar to line 156-157. Can you define a function for this instead of writing the same code twice? Repeating the same code twice is usually a bad sign :)